buildpacks/ruby/src/layers/bundle_install_layer.rs

```diff
@@ -353,9 +355,14 @@ fn bundle_install(env: &Env) -> Result<(), CommandError> {
         ],
     );
 
-    user::log_info(format!("\nRunning command:\n$ {command}"));
+    user::log_info(format!("\nRunning command:\n$ {command_string}"));
 
-    command.stream()?;
+    command
+        .output_and_write_streams(std::io::stdout(), std::io::stderr())
+        .map_err(|error| cmd::annotate_which_problem(error, bundle.into(), env.get("PATH")))
+        .map_err(cmd::os_command_error)
+        .and_then(|output| cmd::check_non_zero(output, cmd::OutputState::AlreadyStreamed))
+        .map_err(|error| CmdError::new(command_string, error))?;
```

Comment:
One of the hard things to solve about the ergonomics here is that I want to be able to chain/compose behavior. In order to get the error command in the error message, there's not a good way to do that since one of the core design goals is to interop natively with Command which returns either an Output or Child when run.

An additional API might be providing a function that takes a Command and then a config object like:

```
cmd::run(Command::new("bundle"), cmd::Config::default())?
```

Or as an extension:

```
Command::new("bundle").run_with(cmd::Config::default))?
```

Have it Return `Result<Option, CmdError>`.

We could still expose a composable API but with something like that, we could have a batteries-included experience. Make it what we want, and if it doesn't meet someone else's needs, they can still use the building blocks.

If we wanted to take functions as config, then the config could include:

- Function to run (Receives a `Command`, returns an `Output`)
- Function to log (Implement a default, but allow an over-ride)
- Function to convert Option into `Result<Option, CmdError>` (default to checking status code)
- Function to apply to std::io::Error (default to WhichProblem)


We have to take into consideration that Command may or may-not contain all the information that is needed in one of these. In addition to program name and arguments, it also takes environment variables and can potentially take a current directory https://doc.rust-lang.org/std/process/struct.Command.html#method.get_envs. I would need to experiment with the API to have more of an opinion.

I like the idea of giving out building blocks, but also having an opinionated batteries-included API.